src/web100srv.c

```diff
@@ -1315,15 +1110,66 @@ int run_test(tcp_stat_agent *agent, Connection *ctl, TestOptions *testopt,
              vars.Timeouts, vars.Sndbuf, vars.MaxRwinRcvd, vars.CurrentCwnd,
              mismatch, bad_cable, totalcnt, refresh);
   }
-  if (ctl->ssl != NULL) {
-    SSL_shutdown(ctl->ssl);
-  } else {
-    shutdown(ctl->socket, SHUT_WR);
-  }
-  /* shutdown(ctlsockfd, SHUT_RDWR); */
+  close_connection(ctl);
+  shutdown(ctl->socket, SHUT_WR);
   return (0);
 }
 
+/** Read a single message sent from the parent server process to the child
+ * process, and perform and necessary communication that message necessitates.
+ * @param parent_pipe where to read the message from
+ * @param ctl the Connection to use for the network communication
+ * @param testopt the parameters of the test
+ * @param t_opt the tests the client supports
+ * @return the message value on success, a negative error code on failure.
+ */
+int read_from_parent(int parent_pipe, Connection *ctl, TestOptions *testopt,
+                     int t_opts) {
+  ParentChildSignal parent_message;
+  int retcode;
+  char buff[16];  // Enough of a buffer to hold an int.
```

Comment:
It's used to hold an int, but also to hold an incoming message (that should be just an int).  It's really just a buffer.  This is likely getting refactored though, so we'll see if the name survives the refactoring.  Comment has been updated.